homeassistant/components/iaqualink/__init__.py

```diff
@@ -228,12 +249,19 @@ def unique_id(self) -> str:
     @property
     def assumed_state(self) -> bool:
         """Return whether the state is based on actual reading from the device."""
-        return not self.dev.system.last_run_success
+        return self.dev.system.online in [False, None]
 
     @property
     def available(self) -> bool:
         """Return whether the device is available or not."""
-        return self.dev.system.online
+        return self.dev.system.online is True
+
+    async def safe_exec(self, awaitable: Awaitable) -> None:
+        """Execute API call while catching service exceptions."""
+        try:
+            await awaitable
+        except AqualinkServiceException as svc_exception:
+            _LOGGER.warning("Aqualink Service Exception raised: %s", svc_exception)
```

Comment:
Does it make sense to trap this and rethrow it as `HomeAssistantError` so the user gets feedback when they try to do something in the UI that fails?